### PR TITLE
ci: optimize workflow

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

- build all branches and pr's
- skip pr when source branch is in local repo

## Context:

Using it in containerbase 🙃 
https://github.com/containerbase/buildpack/blob/683587bfade9f718ca8269e112aed5d1d4077309/.github/workflows/build.yaml#L21

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by HonkingGoose** [comment]
>     * build all branches and pr's

That seems OK.

>     * skip pr when source branch is in local repo

I'm not sure what's the point of this one? Or what it does precisely. Can you please explain this one?

**Comment by viceice** [comment]
> > ```
> > * build all branches and pr's
> > ```
> 
> That seems OK.
> 
> > ```
> > * skip pr when source branch is in local repo
> > ```
> 
> I'm not sure what's the point of this one? Or what it does precisely. Can you please explain this one?

the `push` already builds the commit of the local pr (eg source branch of pr is in same repo as the target branch), so we want save build time by not duplicate builds. As we make sure the pr is always up-to-date before merging we don't need to wait for building the merge-commit, is the source code to build is the same.

**Review by rarkins — APPROVED** [review]

**Pull request merged**
